player/js/utils/common.js

```diff
@@ -122,6 +122,22 @@ const createElementID = (function () {
   };
 }());
 
+function addVectors(a, b) {
+  const result = [];
```

Comment:
This approach doesn't set the `.length` on the `result` array. We should use `.push()` if going in this direction.

However in this case I propose a simpler one-line implementation that also avoids `NaN`s when one of the vectors is shorter:

```javascript
return a.map((_, i) => a[i] + (b[i] || 0));
```

Likewise for the subtraction function.